Support for Unsigned Integer NBITS 1-32

DanielJDufour opened the pull request:
This Pull Requests adds support for geotiffs who use unsigned integer values and NBITS between 1 and 31.

There were a couple things I should mention that surprised me about the GeoTIFF file format.

### Rows breaking on Bytes:
The image data for each row of values breaks on every 8-bit.  That means that if I have a 2x2 unsigned 1bit GeoTIFF, then the values would look like this.
 1 1 0 0 0 0 0 0 1 1 0 0 0 0 0 0
This is why I wrote `getBytesPerRow`

### Need for getUint24, getUint40, getUint48
I initially didn't think I would need this, but it became necessary because sometimes there would only be 3 bytes left in the buffer, which included one more 17-bit pixel value.  Calling the native getUint32 would error because it would be calling for 4 bytes when only 3 bytes are left in the buffer.

I'm definitely open to any advice you can provide on how I can improve this pull request :-)

DanielJDufour commented:
Hi, @constantinius .  Just following up on this.  Is there anything else you need from me on this PR?  Open to your thoughts :-) Thanks!

constantinius commented on src/dataview64.js:
```diff
@@ -46,6 +52,10 @@ export default class DataView64 {
     return this._dataView.getInt16(offset, littleEndian);
   }
 
+  getUint24(offset) {
```
I'm not sure if we actually need this function. Also, the big endian case has to be handled as-well.

constantinius commented on src/dataview64.js:
```diff
@@ -54,6 +64,18 @@ export default class DataView64 {
     return this._dataView.getInt32(offset, littleEndian);
   }
 
+  getUint40(offset, littleEndian) {
```
I'm not sure if we actually need this function. Also, the big endian case has to be handled as-well.

constantinius commented on src/dataview64.js:
```diff
@@ -54,6 +64,18 @@ export default class DataView64 {
+    return (this._dataView.getUint32(offset) << 8) + this._dataView.getUint8(offset + 4);
+  }
+
+  getUint48(offset, littleEndian) {
```
I'm not sure if we actually need this function. Also, the big endian case has to be handled as-well.

constantinius commented on test/data/generate-histograms.py:
```diff
@@ -0,0 +1,51 @@
+from collections import Counter
```
I think it is totally justified to use a Python script here. But could you maybe make it adhere to [PEP 8](https://www.python.org/dev/peps/pep-0008/)?

Also, I see that you are using `rasterio`. This is fine, but maybe add a `requirements.txt` and a small Readme how to use that script?

DanielJDufour commented:
@constantinius , thanks for the feedback! :-) I'll start working on addressing these issues, starting with the easiest stuff.  It may take a month or two to tackle everything.  Thank you again for your guidance.  I appreciate the opportunity to contribute to your awesome project :-)

constantinius commented:
@DanielJDufour 
On the contrary, thanks for contributing!

constantinius commented:
n-bit handling was addressed in a similar PR #135 

Sorry this took so long to settle
